### PR TITLE
nodedev_xml: extend MdevXML with attr elements

### PR DESCRIPTION
**Pull request by smitterl**
Add support for attribute elements of the mediated device capability xml, e.g.:

```
 <attr name='assign_adapter' value='3'/>
 <attr name='assign_domain' value='43'/>
```

### CODE REVIEW

**Comment by smitterl** [comment]
Tested via https://github.com/autotest/tp-libvirt/pull/5040

**Review by cliping — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@chloerh Could you help review it as you also have many experience on this?

File: virttest/libvirt_xml/nodedev_xml.py [hunk 1]
```diff
@@ -152,16 +152,66 @@ class MdevXML(CAPXML):
...
+        Convert an attr tag xml element to an object of AttrXML.
+        """
+        if tag != 'attr':
+            return None    # Don't convert this item
+        newone = AttrXML(virsh_instance=libvirtxml.virsh)
+        newone.xmltreefile = new_treefile
+        return newone
+
+
+class AttrXML(base.LibvirtXMLBase):
```
Comment (chloerh) [inline virttest/libvirt_xml/nodedev_xml.py:197]: According to the xml of <attr>, seems no need to define this class, we could simply define the accessor with 'has_subclass = False' and with shorter marshal functions, if <attr> doesn't have sub-element. What do you think?

Comment (chloerh) [reply virttest/libvirt_xml/nodedev_xml.py:197]: Like this:
```
accessors.XMLElementList('vcpupins', self, parent_xpath='/',
                                 marshal_from=self.marshal_from_vcpupins,
                                 marshal_to=self.marshal_to_vcpupins)
```

Comment (smitterl) [reply virttest/libvirt_xml/nodedev_xml.py]: Thank you for the suggestion. It works well.

Comment (chloerh) [reply virttest/libvirt_xml/nodedev_xml.py]: Thanks for taking my suggestion, btw there's inspekt issue needs to be resovle.

**Review by chloerh — APPROVED** [review]

**Pull request merged**